doc/GitCrashCourse.md

```diff
@@ -0,0 +1,552 @@
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at http://eclipse.org/legal/epl-2.0
+or the Apache License, Version 2.0 which accompanies this distribution
+and is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following Secondary
+Licenses when the conditions for such availability set forth in the
+Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+version 2 with the GNU Classpath Exception [1] and GNU General Public
+License, version 2 with the OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Git crash course
+
+Git is an open-source distributed version control system that the Eclipse
+OMR project uses for tracking code changes. Git is considered a distributed
+version control system because every contributor to a project can have a
+complete copy of a repository and make changes independently. The contributors
+could then request to have their code changes reviewed and merged into the main
+project repository, making it possible for large open-source projects such as
+Eclipse OMR to have many contributors.
+
+You will need to be able to work with git to make code contributions to
+Eclipse OMR. This guide is meant to help you get started and understand
+the project's workflow using git and GitHub by walking you through the process of
+making your first contribution to the Eclipse OMR project hosted on GitHub.
+It is by no means a complete guide for all you can do with git -- there
+are plenty of excellent online resources for that.
+
+# Setting up git and GitHub
+
+This guide will focus on using git on the command line. You may use a git GUI
+application additionally/instead if you so choose.
+
+### Installation
+
+**Linux**
+
+To install git on RPM-based Linux distribution (eg. RHEL), enter the
+following shell commands:
+1. `yum update`
+2. `yum install git`
+
+If you are running a Debian-based distribution (eg. Ubuntu):
+1. `apt-get update`
+2. `apt-get install git`
+
+Note: Depending on your user settings, you may need to prefix the commands
+above with `sudo`.
+
+**macOS**
+
+Launch Terminal.app and install Command Line Tools by entering the
+following command:
+
+```
+xcode-select --install
+```
+
+This command will install git as well as a number of other development
+tools for the command line such as make, clang, svn, etc.
+
+**Windows 10**
+
+There are many ways git can be installed and used on Windows, but the
+simplest way would be downloading the installer from [git-scm download
+page](https://git-scm.com/download/win), and following the installation
+wizard.
+
+---
```

Comment:
I suggest adding something like the following here:

```
### Sign the Eclipse Contributor Agreement (ECA)

Before you can make any code contributions to the public GitHub repository you must
sign the Eclipse Contributor Agreement (ECA).  This is a simple process outlined in the
[Contribution Guidelines](../CONTRIBUTING.md#legal-considerations).  This step is
important because each of your commits must be signed off by the same email you
used to sign the ECA.

---
```